storage/src/key_value/agent/mod.rs

```diff
@@ -142,13 +148,33 @@ impl<S: KeyValueStorage + 'static> Agent<S> {
         Ok(keys.into_iter().map(|(digest, _)| digest).collect())
     }
 
+    fn validate(&mut self, limit: Option<u32>, fix_mode: FixMode) -> bool {
+        let result = if let Some(inner) = mem::take(&mut self.inner) {
+            let (ret, inner) = futures::executor::block_on(inner.validate(limit, fix_mode));
```

Comment:
please be super, super careful with block_on, and try to avoid it if possible. it can create deadlocks very easily.

Reply:
i question the validator needing ownership (why not a mut ref?) -- also if it does need ownership, clone or even better, just take self on `validate` calls since we terminate execution after validating anyways.

Reply:
also why does it need to be async if presumably the only place calling it is sync?

Reply:
this function is intended to only ever be called in some tests (and those have been working well), so we should be fine; good point, of course

Reply:
what do you think about taking ownership then? i.e. `self` target

Reply:
>also why does it need to be async if presumably the only place calling it is sync?
I guess I could use `try_send` in the associated method (which should remove the need for `async`), let me check :+1:

Reply:
>what do you think about taking ownership then? i.e. self target

I could try it out

Reply:
as discussed externally, these changes were currently not viable; they might be possible with planned future changes to the storage